CodeConverter/CSharp/CommonConversions.cs

```diff
@@ -316,22 +316,23 @@ private static string WithDeclarationName(SyntaxToken id, ISymbol idSymbol, stri
             //proxy property
             var baseClassSymbol = GetBaseSymbol(idSymbol, s => s.ContainingType.IsClassType());
             var baseSymbol = GetBaseSymbol(baseClassSymbol, s => true);
-
-            var isCasingDiffOnly = StringComparer.Ordinal.Equals(text, baseClassSymbol.Name) !=
-                                   StringComparer.OrdinalIgnoreCase.Equals(text, baseClassSymbol.Name);
-            var isInterfaceImplRef =
-                baseSymbol.ContainingType.IsInterfaceType() && !(id.Parent is VBSyntax.StatementSyntax);
+            var isInterfaceImplRef = baseSymbol.ContainingType.IsInterfaceType();
             var isDeclaration = isInterfaceImplRef || baseSymbol.Locations.Any(l => l.SourceSpan == id.Span);
+            var isCasingDiffOnly = StringComparer.OrdinalIgnoreCase.Equals(text, baseSymbol.Name) &&
+                                   !StringComparer.Ordinal.Equals(text, baseSymbol.Name);
 
             var isPartial = baseSymbol.IsPartialClassDefinition() || baseSymbol.IsPartialMethodDefinition() ||
                             baseSymbol.IsPartialMethodImplementation();
 
-            if (isInterfaceImplRef && isCasingDiffOnly) {
+            if (isInterfaceImplRef) {
+                if (isCasingDiffOnly && baseClassSymbol.DeclaredAccessibility == Accessibility.Public)
+                    return baseSymbol.Name;
```

Comment:
Minor style point: In general, try to use braces for if statements that don't fit (readably) all on one line.
It doesn't even actually end up that verbose since this code base puts the opening brace on the starting line anyway.